docs/backend-system/building-plugins-and-modules/01-index.md

```diff
@@ -9,4 +9,236 @@ description: Building backend plugins and modules using the new backend system
+using the `catalogProcessingExtensionPoint`:
+
+```ts
+import { createBackendModule } from '@backstage/backend-plugin-api';
+import { catalogProcessingExtensionPoint } from '@backstage/plugin-catalog-node';
+import { MyCustomProcessor } from './MyCustomProcessor';
+
+export const catalogModuleExampleCustomProcessor = createBackendModule({
+  moduleId: 'exampleCustomProcessor',
+  pluginId: 'catalog',
+  register(env) {
+    env.registerInit({
+      deps: {
+        catalog: catalogProcessingExtensionPoint,
+        logger: coreServices.logger,
+      },
+      async init({ catalog }) {
+        catalog.addProcessor(new MyCustomProcessor(logger));
+      },
+    });
+  },
+});
+```
+
+See [the article on naming patterns](../architecture/07-naming-patterns.md) for
+details on how to best choose names/IDs for modules and related backend system
+items.
+
+Notice that we're placing the extension point we want to interact with in the
+`deps` option, while also depending on the logger service at the same time. When
+initializing modules we can depend on both extension points and services
+interchangeably. You can also depend on multiple extension points at once, in
+case the implementation of the module requires it.
+
+It is typically best to keep modules slim and to each only add a single new
+feature. It is often the case that it is better to create two separate modules
+rather than one that provides both features. The one limitation here is that
+modules can not interact with each other and need to be self contained.
+
+### HTTP Handlers
+
+Since modules have access to the same services as the plugin they extend, they
+are also able to register their own HTTP handlers. For more information about
+the HTTP service, see [core services](../core-services/01-index.md). When
+registering HTTP handlers, it is important to try to avoid any future conflict
+with the plugin itself, or other modules. A recommended naming pattern is to
+register the handlers under the `/modules/<module-id>` path, where `<module-id>`
+is the kebab-case ID of the module, for example
+`/modules/example-custom-processor/v1/validators`. In a standard backend setup
+the full path would then be
+`<backendUrl>/api/catalog/modules/example-custom-processor/v1/validators`.
+
+### Database Access
+
+The same applies for modules that perform their own migrations and interact with
+the database. They will run on the same logical database instance as the target
+plugin, so care must be taken to choose table names that do not risk colliding
+with those of the plugin. A recommended naming pattern is `<module name>__<table
+name>`, for example the `@backstage/backend-tasks` package creates tables named
+`backstage_backend_tasks__<table>`. If you use the default Knex migration
+facilities, you will also want to make sure that it uses a similarly prefixed
+lock file for its internal bookkeeping needs, so it does not collide with the
```

Comment:
`migration state table` instead of `lock file`?

Reply:
👍 yeah that reads better, there's a lock table indeed but your wording makes more sense